oauth2-client/src/main/java/org/springframework/security/oauth2/client/authorization/AuthorizationRequestUriBuilder.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.client.authorization;
+
+
+import org.springframework.security.oauth2.core.protocol.AuthorizationRequestAttributes;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * @author Joe Grandja
+ */
+public interface AuthorizationRequestUriBuilder {
```

Comment:
I don't feel we should combine these 2 API's.
`AuthorizationRequestUriBuilder` is used for `authorization_code` and `implicit` grant types whereas `AuthorizationGrantTokenExchanger` is used for `authorization_code`, `client_credentials` and `resource_owner_password` grant types.

We also have only one default implementation for `AuthorizationRequestUriBuilder` that will serve for `authorization_code` and `implicit` grant types. We will need different implementations of `AuthorizationGrantTokenExchanger` for each of  `authorization_code`, `client_credentials` and `resource_owner_password` grant types.